### PR TITLE
[Android] Expose headers parameter to be used by MediaPlayer

### PR DESCRIPTION
**Pull request by fujunwei**
Give the host application a chance to take over the control when a new
media is about to be played in the current XWalkViewInternal.

BUG=XWALK-6770

### CODE REVIEW

**Comment by fujunwei** [comment]
@rakuco @lincsoon @axinging @wuhengzhi PTAL, thanks.

File: media/base/android/java/src/org/chromium/media/MediaPlayerBridge.java [hunk 1]
```diff
@@ -28,17 +28,20 @@
...
 @JNINamespace("media")
 public class MediaPlayerBridge {
-
+    /**
+     * Give the host application a chance to take over the control.
+     */
     public static class ResourceLoadingFilter {
         public boolean shouldOverrideResourceLoading(
-                MediaPlayer mediaPlayer, Context context, Uri uri) {
+                MediaPlayer mediaPlayer, Context context, Uri uri, Map<String, String> headers) {
```
Comment (lincsoon) [inline media/base/android/java/src/org/chromium/media/MediaPlayerBridge.java]: Is `headers` a mandatory parameter? Why don't let the old constructor and the new one coexist?

File: media/base/android/java/src/org/chromium/media/MediaPlayerBridge.java [hunk 2]
```diff
@@ -203,9 +206,9 @@ protected boolean setDataSource(
             headersMap.put("allow-cross-domain-redirect", "false");
         }
         try {
-            if (sResourceLoadFilter != null &&
-                    sResourceLoadFilter.shouldOverrideResourceLoading(
-                            getLocalPlayer(), context, uri)) {
+            if (sResourceLoadFilter != null
+                    && sResourceLoadFilter.shouldOverrideResourceLoading(
```
Comment (lincsoon) [inline media/base/android/java/src/org/chromium/media/MediaPlayerBridge.java]: These two lines don't need to be modified.

File: media/base/android/java/src/org/chromium/media/MediaPlayerBridge.java [hunk 3]
```diff
@@ -203,9 +206,9 @@ protected boolean setDataSource(
...
+                               getLocalPlayer(), context, uri, headersMap)) {
```
Comment (lincsoon) [inline media/base/android/java/src/org/chromium/media/MediaPlayerBridge.java]: Extra spaces at the beginning of the line

File: media/base/android/java/src/org/chromium/media/MediaPlayerBridge.java (hunk 1 above)
Comment (lincsoon) [inline media/base/android/java/src/org/chromium/media/MediaPlayerBridge.java]: Is `headers` a mandatory parameter? Why don't let the method has two interface. If upstream use the old interface in the future, there will be a conflict

Comment (wuhengzhi) [inline media/base/android/java/src/org/chromium/media/MediaPlayerBridge.java]: ```ResourceLoadingFilter``` was added by Crosswalk to resolve media loading issues.

**Pull request closed without merging**